torch/package/package_exporter.py

```diff
@@ -419,6 +431,12 @@ def save_mock_module(self, module_name: str):
         is_package = hasattr(self._import_module(module_name), '__path__')
         self.save_source_string(module_name, _MOCK_IMPL, is_package, dependencies=False)
 
+    def reject_denied_module(self, module_name: str):
+        """Throw an exception containing a message that `module_name` was explicitly blocklisted via
+        `deny` and was still required during packaging.
+        """
+        raise RuntimeError(f"{module_name} was required during packaging but has been explicitly blocklisted")
```

Comment:
similar to before, let's define a specific error class so it can be caught and handled separately by users, DeniedModuleError?